pyscriptjs/tests/test_examples.py

```diff
@@ -0,0 +1,203 @@
+    },
+    "bokeh_interactive": {
+        "file": "bokeh_interactive.html",
+        "pattern": '<div.*?class=\\"bk\\".*?>',
+        "title": "Bokeh Example",
+    },
+    "d3": {
+        "file": "d3.html",
+        "pattern": "<svg.*?>",
+        "title": "d3: JavaScript & PyScript visualizations side-by-side",
+    },
+    "folium": {"file": "folium.html", "pattern": "<iframe srcdoc=", "title": "Folium"},
+    "hello_world": {
+        "file": "hello_world.html",
+        "pattern": "\\d+/\\d+/\\d+, \\d+:\\d+:\\d+",
+        "title": "PyScript Hello World",
+    },
+    "matplotlib": {
+        "file": "matplotlib.html",
+        "pattern": "<img src=['\"]data:image",
+        "title": "Matplotlib",
+    },
+    "numpy_canvas_fractals": {
+        "file": "numpy_canvas_fractals.html",
+        "pattern": "<div.*?id=['\"](mandelbrot|julia|newton)['\"].*?>",
+        "title": "Visualization of Mandelbrot, Julia and "
+        "Newton sets with NumPy and HTML5 canvas",
+    },
+    "panel": {
+        "file": "panel.html",
+        "pattern": "<div.*?class=['\"]bk-root['\"].*?>",
+        "title": "Panel Example",
+    },
+    "panel_deckgl": {
+        "file": "panel_deckgl.html",
+        "pattern": "<div.*?class=['\"]bk-root['\"].*?>",
+        "title": "PyScript/Panel DeckGL Demo",
+    },
+    "panel_kmeans": {
+        "file": "panel_kmeans.html",
+        "pattern": "<div.*?class=['\"]bk-root['\"].*?>",
+        "title": "Pyscript/Panel KMeans Demo",
+    },
+    "panel_stream": {
+        "file": "panel_stream.html",
+        "pattern": "<div.*?class=['\"]bk-root['\"].*?>",
+        "title": "PyScript/Panel Streaming Demo",
+    },
+    "repl": {"file": "repl.html", "pattern": "<py-repl.*?>", "title": "REPL"},
+    "repl2": {
+        "file": "repl2.html",
+        "pattern": "<py-repl.*?>",
+        "title": "Custom REPL Example",
+    },
+    "simple_clock": {
+        "file": "simple_clock.html",
+        "pattern": "\\d+/\\d+/\\d+, \\d+:\\d+:\\d+",
+        "title": "Simple Clock Demo",
+    },
+    "todo": {
+        "file": "todo.html",
+        "pattern": "<input.*?id=['\"]new-task-content['\"].*?>",
+        "title": "Todo App",
+    },
+    "todo_pylist": {
+        "file": "todo-pylist.html",
+        "pattern": "<input.*?id=['\"]new-task-content['\"].*?>",
+        "title": "Todo App",
+    },
+    "toga_freedom": {
+        "file": "toga/freedom.html",
+        "pattern": "<(main|div).*?id=['\"]toga_\\d+['\"].*?>",
+        "title": ["Loading...", "Freedom Units"],
+    },
+    "webgl_raycaster_index": {
+        "file": "webgl/raycaster/index.html",
+        "pattern": "<canvas.*?>",
+        "title": "Raycaster",
+    },
+}
+
+
+@pytest.mark.parametrize("example", EXAMPLES)
+def test_examples(example, http_server):
+
+    base_url = http_server
+    example_path = urljoin(base_url, TEST_PARAMS[example]["file"])
+
+    # Invoke playwright
+    with sync_playwright() as p:
+        browser = p.chromium.launch()
```

Comment:
Its fine for the first pass would be nice to make it browser agnostics so can be tested against multiple browsers.